### PR TITLE
Refactor Params -> ClientOptions and logging

### LINKED ISSUE #15: API changes
Don't action yet, but take a look at https://github.com/ably/ably-js/issues/33 where we are proposing a final set of API changes before we roll out stable versions.

### PR DESCRIPTION
**Pull request by rjeczalik**
Continuation of #13.

Closes #15.

### CODE REVIEW

File: ably/testutil/test_app.go [hunk 1]
```diff
@@ -36,14 +36,14 @@ type testAppChannels struct {
...
-	AppID      string             `json:"appId,omitempty"`
+	ApiID      string             `json:"appId,omitempty"`
 	Keys       []testAppKey       `json:"keys"`
 	Namespaces []testAppNamespace `json:"namespaces"`
 	Channels   []testAppChannels  `json:"channels"`
 }
 
 type App struct {
-	Params ably.Params
+	Params ably.ClientOptions
```
Comment (kouno) [inline ably/testutil/test_app.go]: You should rename `Params` too then.

File: ably/auth.go [hunk 2]
```diff
@@ -71,20 +71,20 @@ func (t *TokenRequest) Sign(secret string) {
 }
 
 type Auth struct {
-	Params
+	ClientOptions
 	client *RestClient
 }
 
-func NewAuth(params Params, client *RestClient) *Auth {
+func NewAuth(params ClientOptions, client *RestClient) *Auth {
```
Comment (kouno) [inline ably/auth.go]: Rename `params` to `clientOptions`.

File: ably/testutil/test_app.go (hunk 1 above)
Comment (rjeczalik) [inline ably/testutil/test_app.go]: Ops, done.

File: ably/auth.go (hunk 2 above)
Comment (rjeczalik) [inline ably/auth.go]: Done.

**Comment by kouno** [comment]
Sweet! Thanks.

**Pull request merged**